src/renderer/services/dex.js

```diff
@@ -619,10 +624,26 @@ export default {
           depositPromises.push(this.depositAsset(d.assetId, d.quantityToDeposit.toNumber()));
         });
 
+        const watchInterval = setInterval(() => {
+          let waiting = false;
+          order.deposits.forEach((deposit) => {
+            const holding = neo.getHolding(deposit.assetId);
+            if (holding.contractBalance.isLessThan(deposit.quantityRequired)) {
+              waiting = true;
+            }
+          });
+
+          if (waiting === false) {
+            clearInterval(watchInterval);
+            resolve(order);
+          }
+        }, intervals.BLOCK);
+
         Promise.all(depositPromises)
           .then(() => {
             store.dispatch('fetchHoldings', {
               done: () => {
+                clearInterval(watchInterval);
```

Comment:
Is this redundant with the clearInterval called when on line 637?

Reply:
No they're both necessary, the one lower may find the deposit succeeded first. The top is more of a failsafe